### PR TITLE
Fix crash on mismatch component lengths

### LINKED ISSUE #3270: new api crashes when archetype instance count mismatch with component arrays
Simple crash repro from python
`rr2.log("crash", rr2.Points2D([0, 0], instance_keys=[0, 1]))`

### PR DESCRIPTION
**Pull request by emilk**
* Closes #3270

Adds `_or_panic` suffix to functions that panics on error and stops using them in a bunch of places.

We cannot do `.unwrap()` in functions with innocuous names. We will get crashes. Use `Result`.

Next step is to move away from these horrible `_or_panic` functions, because we don't want the process to die.

If you add a `.unwrap()` explain _why it can't happen_.
Any naked (unexplained) `.unwrap()` in our code is an implicit TODO that should be fixed.
Ideally we should enable `clippy::unwrap_used` on our code-base.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3406) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3406)
- [Docs preview](https://rerun.io/preview/7048df0b562c855601e702f74cc37f1221f1c933/docs) 
- [Examples preview](https://rerun.io/preview/7048df0b562c855601e702f74cc37f1221f1c933/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

**Comment by emilk** [comment]
* Related: https://github.com/rerun-io/rerun/issues/3408

File: crates/re_log_types/src/data_table.rs [hunk 1]
```diff
@@ -221,18 +221,18 @@ impl std::ops::DerefMut for TableId {
...
+///     DataRow::try_from_cells1(RowId::random(), "b", timepoint(1, 2), num_instances, colors)
 /// };
 /// let row2 = {
 ///     let num_instances = 1;
 ///     let colors: &[_] = &[MyColor::from_rgb(255, 255, 255)];
 ///     let labels: &[_] = &[Label("hey".into())];
-///     DataRow::from_cells2(RowId::random(), "c", timepoint(2, 1), num_instances, (colors, labels))
+///     DataRow::try_from_cells2(RowId::random(), "c", timepoint(2, 1), num_instances, (colors, labels))
 /// };
 /// let table = DataTable::from_rows(table_id, [row0, row1, row2]);
```
Comment (Wumpf) [inline crates/re_log_types/src/data_table.rs:237]: doc string code is broken here, this passes in the results as rows

**Review by Wumpf — APPROVED** [review]
it's a bit confusing which methods have an `or_panic` version and which ones don't. Fairly sure you sometimes use the try_ version only to unwrap it then. Which hints that the try/or_panic version might be wrong to begin with

**Comment by emilk** [comment]
Yes, I think we should strive to remove all `_or_panic` functions.

**Comment by emilk** [comment]
Ok, I removed the `_or_panic` things I introduced, and all errors are now correctly propagated!

**Pull request merged**